# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/build/XPCOMInit.cpp
# Commit: 3e501ab00f18
# Full Hash: 3e501ab00f1855046db1f8db6be58fb35f7c6adc
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-04-07 03:19:44
# Regressor Bug: 1695580
# File Overlap Count: 1
# Description:
#   Bug 1695580 - In xpcom, cancel pending DelayedRunnable timers on shutdown. r=KrisWright
#   
#   Because DelayedRunnables are fire-and-forget, there is no way for a targeted
#   EventTarget to clean them up on shutdown. Thus if a timer fires after
#   EventTarget shutdown it will fail to dispatch the timer event, and avoid
# ==============================================================================

diff -r 2a88ea548a86 -r 3e501ab00f18 xpcom/build/XPCOMInit.cpp
--- a/xpcom/build/XPCOMInit.cpp	Tue Apr 06 20:15:10 2021 +0000
+++ b/xpcom/build/XPCOMInit.cpp	Tue Apr 06 20:15:11 2021 +0000
@@ -621,6 +621,7 @@
 
     mozilla::AppShutdown::AdvanceShutdownPhase(
         mozilla::ShutdownPhase::XPCOMShutdownThreads);
+    nsThreadManager::get().CancelBackgroundDelayedRunnables();
     gXPCOMThreadsShutDown = true;
     NS_ProcessPendingEvents(thread);
 